File: 자료구조/Python/백준-데큐-9093.py
from sys import stdin

def main():
    N = int(stdin.readline())

    stack = []

    for i in range(N):
        sentence = stdin.readline()
        for alpha in sentence:
            if alpha == ' ':
                while stack:
                    print(stack.pop(), end="")
                print(alpha, end="")
            elif alpha == '\n':
                continue
            else:
                stack.append(alpha)

        while stack:
            print(stack.pop(), end="")
        print("")
if __name__ == "__main__":
    main()

File: 자료구조/C++/백준-데큐-9093.cpp
# include <iostream>
# include <stack>
# include <string>

using namespace std;

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    int N;
    cin >> N;
    cin.ignore(); // 입력 버퍼 비우기
    string sentence;

    stack<char> st;

    for (int i = 0; i < N; i++)
    {
        getline(cin, sentence);
        
        for (char c : sentence) 
        {
            if (c == ' ')
            {
                while (!st.empty())
                {
                    cout << st.top();
                    st.pop();
                }
                cout << ' ';
            }
            else if (isalnum(c))
            {
                st.push(c);
            }
        }
        while (!st.empty())
        {
            cout << st.top();
            st.pop();
        }
        cout << endl;
    }

    return 0;
}